servers/src/main/java/tachyon/worker/block/TieredBlockStore.java

```diff
@@ -0,0 +1,432 @@
+    TempBlockMeta tempBlockMeta = mMetaManager.getTempBlockMeta(blockId);
+    return new LocalFileBlockWriter(tempBlockMeta);
+  }
+
+  @Override
+  public BlockReader getBlockReader(long userId, long blockId, long lockId) throws IOException {
+    mLockManager.validateLockId(userId, blockId, lockId);
+    BlockMeta blockMeta = mMetaManager.getBlockMeta(blockId);
+    return new LocalFileBlockReader(blockMeta);
+  }
+
+  @Override
+  public TempBlockMeta createBlockMeta(long userId, long blockId, BlockStoreLocation location,
+      long initialBlockSize) throws IOException {
+    TempBlockMeta tempBlock;
+    mEvictionLock.readLock().lock();
+    try {
+      tempBlock = createBlockMetaNoLock(userId, blockId, location, initialBlockSize);
+    } finally {
+      mEvictionLock.readLock().unlock();
+    }
+    return tempBlock;
+  }
+
+  @Override
+  public BlockMeta getBlockMeta(long userId, long blockId, long lockId) throws IOException {
+    mLockManager.validateLockId(userId, blockId, lockId);
+    return mMetaManager.getBlockMeta(blockId);
+  }
+
+  @Override
+  public void commitBlock(long userId, long blockId) throws IOException {
+    mEvictionLock.readLock().lock();
+    TempBlockMeta tempBlockMeta = mMetaManager.getTempBlockMeta(blockId);
+    try {
+      commitBlockNoLock(userId, blockId, tempBlockMeta);
+      for (BlockStoreEventListener listener : mBlockStoreEventListeners) {
+        listener.onCommitBlock(userId, blockId, tempBlockMeta.getBlockLocation());
+      }
+    } finally {
+      mEvictionLock.readLock().unlock();
+    }
+  }
+
+  @Override
+  public void abortBlock(long userId, long blockId) throws IOException {
+    mEvictionLock.readLock().lock();
+    try {
+      abortBlockNoLock(userId, blockId);
+      for (BlockStoreEventListener listener : mBlockStoreEventListeners) {
+        listener.onAbortBlock(userId, blockId);
+      }
+    } finally {
+      mEvictionLock.readLock().unlock();
+    }
+  }
+
+  @Override
+  public void requestSpace(long userId, long blockId, long moreBytes) throws IOException {
+    // TODO: Change the lock to read lock and only upgrade to write lock if necessary
+    mEvictionLock.writeLock().lock();
+    try {
+      TempBlockMeta tempBlockMeta = mMetaManager.getTempBlockMeta(blockId);
+      BlockStoreLocation location = tempBlockMeta.getBlockLocation();
+      freeSpaceInternal(userId, moreBytes, location);
+      // Increase the size of this temp block
+      mMetaManager.resizeTempBlockMeta(tempBlockMeta, tempBlockMeta.getBlockSize() + moreBytes);
+    } finally {
+      mEvictionLock.writeLock().unlock();
+    }
+  }
+
+  @Override
+  public void moveBlock(long userId, long blockId, BlockStoreLocation newLocation)
+      throws IOException {
+    mEvictionLock.readLock().lock();
+    try {
+      long lockId = mLockManager.lockBlock(userId, blockId, BlockLockType.WRITE);
+      BlockMeta blockMeta = mMetaManager.getBlockMeta(blockId);
```

Comment:
Should this also be in the try clause since it can throw an exception and leak the block lock?